File: JAVA/Insertion Sort for Singly Linked List.java
class Solution
{
    public static Node insertionSort(Node head)
    {
        ArrayList<Integer> a = new ArrayList<>();
        Node temp = head;
        while(temp != null){
            a.add(temp.data);
            temp = temp.next;
        }
        Collections.sort(a);
        temp = head;
        int i = 0;
        while(temp != null){
            temp.data = a.get(i);
            i++;
            temp = temp.next;
        }
        return head;
    }
}
 

File: C++/Insertion Sort for Singly Linked List.cpp
class Solution
{
    public:
    Node* insertionSort(struct Node* head)
    {
        //code here
        if (!head || head->next == NULL)
            return head;
        Node* extra = new Node(-1);
        extra->next = head;
        
        Node * curr = head->next;
        Node * prev = head;
        while(curr != NULL){
            Node * temp = extra;
            if (prev->data < curr->data){
                curr = curr->next;
                prev = prev->next;
                continue;
            }
            while(temp->next->data < curr->data){
                temp = temp->next;
            }
            prev->next = curr->next;
            curr->next = temp->next;
            temp->next = curr;
            curr = prev->next;

        }
        return extra->next;
    }
    
};

 
